include/zephyr/net/buf.h

```diff
@@ -2414,6 +2414,24 @@ size_t net_buf_append_bytes(struct net_buf *buf, size_t len,
 			    const void *value, k_timeout_t timeout,
 			    net_buf_allocator_cb allocate_cb, void *user_data);
 
+/**
+ * @brief Compare data with a net_buf's content
+ *
+ * @details Compare data with a content of a net_buf. Provide information about
+ * the number of bytes matching between both. If needed, traverse
+ * through multiple buffer fragments.
+ *
+ * @param buf Network buffer
+ * @param offset Starting offset to compare from
+ * @param data Data buffer for comparison
+ * @param data_len Length of the data buffer
+ * @param len Number of bytes to compare
+ *
+ * @return The number of bytes compared before the first difference.
+ */
+size_t net_buf_data_cmp(struct net_buf *buf, size_t offset, void *data,
+			size_t data_len, size_t len);
```

Comment:
Couldn't you just remove the `data_len` parameter? That seems unnecessary. The only place you use it is in the beginning of the function to test `(len > data_len)`. Ensuring that actual valid data behind the `data` pointer is between `0` and `len` should be the responsibility of the caller.

Reply:
@jhedberg I have removed the parameter